cmd/dev.go

```diff
@@ -0,0 +1,158 @@
+//go:build !windows
+
+package cmd
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/skatteetaten/ao/pkg/auroraconfig"
+	"github.com/skatteetaten/ao/pkg/deploymentspec"
+	"github.com/skatteetaten/ao/pkg/prompt"
+	architect "github.com/skatteetaten/architect/v2/pkg/build"
+	architectConfig "github.com/skatteetaten/architect/v2/pkg/config"
+	"github.com/spf13/cobra"
+	"os"
+	"strings"
+)
+
+const OpenshiftRegistryFormat = "default-route-openshift-image-registry.apps.%s.paas.skead.no"
+
+var (
+	devCmd = &cobra.Command{
+		Use:         "dev",
+		Short:       "Perform dev operations",
+		Annotations: map[string]string{"type": "development"},
+	}
+
+	rolloutCmd = &cobra.Command{
+		Use:   "rollout <applicationdeploymentRef> <Leveransepakke file>",
+		Short: "Rollout leveransepakke to openshift for development type",
+		RunE:  Rollout,
+	}
+)
+
+func init() {
+	devCmd.AddCommand(rolloutCmd)
+	RootCmd.AddCommand(devCmd)
+}
+
+func Rollout(cmd *cobra.Command, args []string) error {
+	if len(args) != 2 {
+		return cmd.Usage()
+	}
+
+	leveransePakkeFile := args[1]
+	if err := verifyFileExists(leveransePakkeFile); err != nil {
+		return err
+	}
+
+	fileNames, err := DefaultAPIClient.GetFileNames()
+	if err != nil {
+		return err
+	}
+
+	search := args[0]
+
+	matches, err := getFuzzyMatchedResult(search, fileNames)
+	if err != nil {
+		return err
+	}
+
+	spec, err := getSpec(matches)
+	if err != nil {
+		return err
+	}
+
+	deployType := spec.Get("type")
+	if deployType != "development" {
+		return fmt.Errorf("You need to specify type=development in AuroraConfig. Current type=%s", deployType)
+	}
+
+	conf := getArchitectBuildConfiguration(spec, leveransePakkeFile)
+
+	cmd.Println("Building image.. If you wish to see the details you can run the command with `-l info` flag.")
+	architect.BuildBinary(conf)
+
+	return nil
+}
+
+func getSpec(matches []string) (deploymentspec.DeploymentSpec, error) {
+	allSpecs, err := DefaultAPIClient.GetAuroraDeploySpec(matches, !flagNoDefaults, flagIgnoreErrors)
+	if err != nil {
+		return nil, err
+	}
+
+	return allSpecs[0], nil
```

Comment:
Siden man kan få flere tilslag fra fuzzy-matchingen blir denne uforutsigbar og litt skummel.  getSpec bør feile dersom det er mer enn én match.  (Dette blir litt som for ao deploy med version-flagg eller ao get spec. Begge de kommandoene bruker også fuzzy-matching og skal feile dersom det er mer enn én match)

Jeg ser at denne er "sikret" av at getFuzzyMatchedResult feiler om det ikke er en eksakt match, men da bør den returnere kun en verdi.